packages/repository-json-schema/src/build-schema.ts

```diff
@@ -486,6 +486,19 @@ export function modelToJsonSchema<T extends object>(
       continue;
     }
 
+    if (meta?.settings?.hiddenProperties?.includes(p)) {
+      debug(
+        'Property % is hidden by model hiddenProperties settings %s',
+        meta?.settings?.hiddenProperties?.includes(p),
+      );
+      continue;
+    }
+
+    if (meta.properties[p].hidden) {
```

Comment:
If a model property has a hidden property, it is added to meta.settings.hiddenProperties. You probably don't need this check. The first condition should be enough.

Reply:
I tried removing the check, and I'm not seeing the behavior you described.
The 'HiddenProperty' tests are now failing with the properties in the schema not removed. I tried debugging the unit test `excludes one property when the property setting is set to hidden` and I'm seeing the `SingleHiddenPropertyProduct` having no `meta.settings` sub-properties available

![image](https://github.com/user-attachments/assets/fad907c1-3987-4003-90f5-135cab917768)

Reply:
It looks like that behavior only happens in `packages/repository/src/decorators/model.decorator.ts` / `buildModelDefinition`, which is run once on the start, but when the `modelToJsonSchema` is called, the `ModelMetadataHelper.getModelMetadata` is unable to locate the cached model, and rebuilds the model meta (without the behavior from the `buildModelDefinition` that adds the hidden properties to the model's `settings.hiddenProperties`)

Reply:
Hi, @taylorcoffelt. You are right. meta.settings.hiddenProperties is not populated with hidden properties here.

This change not only hides the properties from the response, it hides from the request body example too - which it shouldn't as we can post with hidden fields.

Secondly, the filters still show the hidden fields.

You can hide properties in the filter by filtering hidden fields [here](https://github.com/loopbackio/loopback-next/blob/3cf7046ee1a6b1b3509b70ce4bd82af56d9eba71/packages/repository-json-schema/src/filter-json-schema.ts#L214).

```
const hiddenProperties = modelCtor.definition.settings.hiddenProperties || [];
const originalProps = Object.keys(modelCtor.definition.properties);
const properties = originalProps.filter(prop => !hiddenProperties.includes(prop));
```

While you can remove the hidden properties from the responses in `/rest-crud/src/crud-rest.controller.ts`.

```
    @get('/{id}', {
      ...response.model(200, `${modelName} instance`, modelCtor, {
        includeRelations: true,
      }),
    })
 ```
 
 We can pass exclude as an array along with includeRelations.